### ISSUE TITLE
Registration Mark Issues on Silhouette Cameo 3

### ISSUE THREAD

**Issue by sxwid** [fablabnbg/inkscape-silhouette#82]
I successfully started using your Extension. Thanks already 1000 time I dont need this lousy Silhouette Studio anymore! Vinyl cutting is already perfect (after using the masterbranch - maybe release a 1.20?)
But I am facing issues with the registration marks. So far I am using `registration-marks-cameo-silhouette-a4-maxi.svg`. But I have odd offsets of the cuts, which are changing over the whole document. It is getting a little better by adjusting the X- and Y-Offset in the main Menu. Might it be this is due to this dpi problem? I dont really see what the magic Numbers in the Regmark setting make - can anyone enlighten me? Thanks!

**Comment by prosoftJit** [comment]
Im with same problem. Cutting size scales over page when i use registration marks. When i dont use it, cutting goes very well...
My cameo 3 reads the registration marks as well.

**Comment by MagpieRH** [comment]
Have had similar issues with reg marks - I've found the measurements on my Portrait 2 need to be taken from the top edge of the mat (rather than the edge of the sticky surface) for them to be found, and even then it cuts in strange ways.
I have only tried with simple shapes, as a test, but the 2 boxes I had laid out  missed the target, and only cut one line. Second attempt, they again cut one line, but slightly offset from the first.

**Comment by sxwid** [comment]
Correct, you need to stick it to the absolute corner of the sticky mat. Also I am working with these settings, which fit my needs for most cases. Still I get offset on the bottom and right side of the page, but I am not sure if this is an issue of the extension or just a hardware limit of the cameo. Below the accuracy I reach in different areas of an a4 sheet.
![silhouette](https://user-images.githubusercontent.com/13117378/80915003-9762f880-8d4f-11ea-8f41-acccad8611c5.png)

**Comment by ktosiek** [comment]
I have a similar problem. Playing with X/Y mark distance helped a bit, I had to make them 0.1mm lower then expected. But there's a 0.1-0.2mm change in offset in both X and Y between top left and bottom right.

**Comment by ktosiek** [comment]
How are the registration marks actually used? ~I see there's a command sent to the plotter, but it seems the code ignores the response?~ looks like plotter only sends an "ok!" response to the auto regmark command.

**Comment by EtherGraf** [comment]
The logic behind the regmark command seems to be as following: 

The plotter ist told (values from above axample)
"Search for the regmarks. Search the origin postion x,y. (not sure, I believe, this is hardcoded). Expect the real physical distances 190 mm and 277 mm and search the corners in this region. If there are some differences, then do the fine calibration of the paper (translation, rotation) byself."
So there is no need to evaluate any response.

I do not expect the size to be corrected too, only translation and rotation.

**Comment by ktosiek** [comment]
I've found a solution to my problem: inkscape-silhouette should scale all paths (and offsets too, I think?) according to distances stored in the plotter. I think the `FB` command only saves the scaling factors, but they don't affect the cut.

What I've done:
1. I've went through calibration in Silhouette Studio (which pretty much cuts 2 pairs of markers and asks you to measure distance between them), and I've seen Silhouette Studio send an `FB50,0` command. This did not affect cuts done with inkscape-silhouette.
2. I've changed the code in `SilhouetteCameo.plot()` to multiply points in all paths by x=1.002, y=1.003, and it made cuts precise over the whole page. (Values obtained experimentally, I think those would be `FB30,10` - I'm not sure if my measurements from step 1. were good).

I'll try to prepare a PR for setting the scaling factors in the UI, and with a bit of luck - for using the ones saved in the machine.

Is there any knowledge on how the scaling should work? I don't see anything in Robocut, and Inkcut only supports 3-4 GPGL commands.

**Comment by EtherGraf** [comment]
In https://github.com/fablabnbg/inkscape-silhouette/blob/master/doc/commands.md are some hints for the FA command:

Unknown: 2 five digit numbers. Probably machine stored calibration factors of carriage and roller. (carriage, roller / unit 1/100% i.e. 0.0001)

**Comment by ktosiek** [comment]
Turns out my approach did not work, I've just moved the scaling issues from the right (far from start) side of paper to the left side.

**Comment by ktosiek** [comment]
I'm starting to believe scaling is done by the plotter, and registration marks are used instead of the scaling settings when available.

This is all done with the Studio:
1. I've done 2 cuts with regmarks, one with no scaling and one with `FB49,0` - both cut almost identically (and both way off at bottom right - SS seems to show the "bottom right problem" too). Both show same commands in a USB capture.
2. Then I've done 2 cuts without regmarks, same file, one with no scaling (`FB0,0`), and one with some big scaling in Y axis (around 1.9%, I'm not sure exactly right now). The cuts were different, but the captured commands still stayed the same in both cuts.

**Comment by neduz** [comment]
I've tried using registration marks with a Cameo 3, but the result was that a square  (in the top half of the paper) came out skewed, looking like a parallelogram( one vertical side being ~5mm higher than the other).
Are there any patterns you recommend to test with to debug this on Cameo 3?
